src/fhirpath.js

```diff
@@ -584,16 +584,22 @@ var evaluate = function(resource, path, context) {
 };
 
 /**
- *  Returns a function that takes a resource and returns the result of
- *  evaluating the given FHIRPath expression on that resource.  The advantage
- *  of this function over "evaluate" is that if you have multiple resources,
- *  the given FHIRPath expression will only be parsed once.
+ *  Returns a function that takes a resource and an optional context hash (see
+ *  "evaluate"), and returns the result of evaluating the given FHIRPath
+ *  expression on that resource.  The advantage of this function over "evaluate"
+ *  is that if you have multiple resources, the given FHIRPath expression will
+ *  only be parsed once.
  * @param path the FHIRPath expression to be parsed.
- * @param {object} context - a hash of variable name/value pairs.
+ * @param {object} (deprecated) context - a hash of variable name/value pairs.  This is
+ *  optional now, and is deprecated, because it was probably a mistake.  Instead
+ *  of passing in this hash of variables here, pass it to the returned function
+ *  as a second argument.
```

Comment:
Maybe also a note on what happens if a context is passed in here, and while calling the returned function, a context, potentially a different one, is also passed in? E.g., two contexts are merged (with override), or one context completely replaces the other?

Reply:
Good point.  Changed.